### ISSUE TITLE
False positive (Suspicious assignment && Suspicious loop)

### ISSUE THREAD

**Issue by orklah** [kalessil/phpinspectionsea#934]
Hi! 

Two false positive to report:
```php
$array[__LINE__] = 'foo';
$array[__LINE__] = 'bar';
```

triggers Suspicious assignment > immediatly overriden instruction

```php
$authorize_single_element = true;
$array = array_fill(0, random_int(1,2), 'foo');
if((count($array) === 1 && $authorize_single_element) || count($array) > 1){
    foreach($array as $val){
        echo $val;
    }
}
```

triggers Suspicious loop > suspicious parent condition

Thank you for your amazing work!

**Comment by kalessil** [comment]
Thank you for reporting @orklah, I'll check this.

**Comment by kalessil** [comment]
1) Fixed
2) No chance to fix this - suppression is recommended
